filesystem/iso9660.ksy

```diff
@@ -5,269 +5,589 @@ meta:
+        doc-ref: ecma-119 8.4.24
+        type: volume_text_37
+      - id: bibliographic_file_identifier
+        doc-ref: ecma-119 8.4.25
+        type: volume_text_37
+      - id: volume_creation_date_and_time
+        doc-ref: ecma-119 8.4.26
+        type: datetime_long
+      - id: volume_modification_date_and_time
+        doc-ref: ecma-119 8.4.27
+        type: datetime_long
+      - id: volume_expiration_date_and_time
+        doc-ref: ecma-119 8.4.28
+        type: datetime_long
+      - id: volume_effective_date_and_time
+        doc-ref: ecma-119 8.4.29
+        type: datetime_long
       - id: file_structure_version
-        type: u1
-      - id: unused4
-        type: u1
-      - id: application_area
-        size: 512
-    instances:
-      path_table:
-        pos: lba_path_table_le * _root.sector_size
-        size: path_table_size.le
-        type: path_table_le
-  dir_entries:
-    seq:
-      - id: entries
-        type: dir_entry
-        repeat: until
-        repeat-until: _.len == 0
-  dir_entry:
+        doc-ref: ecma-119 8.4.31
+        type: s1
+  supplementary_volume:
+    doc-ref: ecma-119 8.5
     seq:
-      - id: len
-        type: u1
-      - id: body
-        type: dir_entry_body
-        size: len - 1
-        if: len > 0
-  dir_entry_body:
-    seq:
-      - id: len_ext_attr_rec
-        type: u1
-      - id: lba_extent
+      - id: volume_flags_reserved
+        doc-ref: ecma-119 8.5.3 b1-b7
+        type: b7
+      - id: volume_flags_not_iso2375
+        doc-ref: ecma-119 8.5.3 b0
+        type: b1
+      - id: system_identifier
+        doc-ref: ecma-119 8.5.4
+        type: volume_text_32
+      - id: volume_identifier
+        doc-ref: ecma-119 8.5.5
+        type: volume_text_32
+      - id: unused01
+        doc-ref: ecma-119 8.5
+        size: 0x8
+      - id: volume_space_size
+        doc-ref: ecma-119 8.5
         type: u4bi
-      - id: size_extent
+      - id: escape_sequences
+        doc-ref: ecma-119 8.5.6
+        size: 0x20
+      - id: volume_set_size
+        doc-ref: ecma-119 8.5
+        type: u2bi
+      - id: volume_sequence_number
+        doc-ref: ecma-119 8.5
+        type: u2bi
+      - id: logical_block_size
+        doc-ref: ecma-119 8.5
+        type: u2bi
+      - id: path_table_size
+        doc-ref: ecma-119 8.5.7
         type: u4bi
-      - id: datetime
-        type: datetime
-      - id: file_flags
+      - id: occurrence_of_type_l_path_table
+        doc-ref: ecma-119 8.5.8
+        type: u4le
+      - id: optional_occurrence_of_type_l_path_table
+        doc-ref: ecma-119 8.5.9
+        type: u4le
+      - id: occurrence_of_type_m_path_table
+        doc-ref: ecma-119 8.5.10
+        type: u4le
+      - id: optional_occurrence_of_type_m_path_table
+        doc-ref: ecma-119 8.5.11
+        type: u4le
+      - id: directory_record_for_root_directory
+        doc-ref: ecma-119 8.5.12
+        type: directory_record_head
+        size: 0x22
```

Comment:
Yet another magic constant. Can we eliminate it?

Reply:
Looks like we have a reusable piece which can be moved into a substruct, don't we?

Reply:
It already is, a directory_record_head, unless I don't understand you correctly